cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerSetupService.java

```diff
@@ -534,7 +539,15 @@ private ApiClusterTemplate getCmTemplate(TemplatePreparationObject templatePrepa
 
     private void configureCmMgmtServices(TemplatePreparationObject templatePreparationObject, String sdxCrn, Telemetry telemetry,
             String sdxContextName, ProxyConfig proxyConfig) throws ApiException {
-        Optional<ApiHost> optionalCmHost = getCmHost(templatePreparationObject, apiClient);
+        Optional<ApiHost> optionalCmHost;
+        if (null != templatePreparationObject.getStackType()
+            && templatePreparationObject.getStackType().equals(StackType.DATALAKE)
+            && isVersionNewerOrEqualThanLimited(templatePreparationObject.getBlueprintView().getVersion(), CLOUDERA_STACK_VERSION_7_2_17)
+            && containsIgnoreCase(templatePreparationObject.getBlueprintView().getBlueprintText(), SDX_ENTERPRISE_DATALAKE_TEXT)) {
+            optionalCmHost = getAuxiliaryHost(templatePreparationObject, apiClient);
+        } else {
+            optionalCmHost = getCmHost(templatePreparationObject, apiClient);
+        }
```

Comment:
I would add some logs to both branch so we would see from the logs which path the code took